web/src/main/react-dashboard/src/components/TestRunView.js

```diff
@@ -49,563 +42,570 @@ import {FAIL, SUCCESS, ERROR, PENDING, RUNNING, HTTP_UNAUTHORIZED, LOGIN_URI, TE
  */
 class TestRunView extends Component {
 
-  constructor(props) {
-    super(props);
-    this.state = {
-      testScenarioSummaries: [],
-      scenarioTestCaseEntries: [],
-      testSummaryLoadStatus: PENDING,
-      logContent: "",
-      logDownloadStatus: PENDING,
-      isLogTruncated: false,
-      inputStreamSize: "",
-      showLogDownloadErrorDialog: false
-    };
-  }
+    constructor(props) {
+        super(props);
+        this.state = {
+            testScenarioSummaries: [],
+            scenarioTestCaseEntries: [],
+            testSummaryLoadStatus: PENDING,
+            logContent: "",
+            logDownloadStatus: PENDING,
+            isLogTruncated: false,
+            inputStreamSize: "",
+            showLogDownloadErrorDialog: false,
+            currentInfra: null
+        };
+    }
 
-  componentDidMount() {
-    const testScenarioSummaryUrl = TESTGRID_CONTEXT + '/api/test-plans/test-summary/' +
-      this.props.active.reducer.currentInfra.testPlanId;
-    const logTruncatedContentUrl = TESTGRID_CONTEXT + '/api/test-plans/log/' +
-      this.props.active.reducer.currentInfra.testPlanId + "?truncate=" + true;
+    componentDidMount() {
+        let currentInfra = {};
+        let currentUrl = window.location.href.split("/");
+        currentInfra.relatedProduct = currentUrl[currentUrl.length - 4];
+        currentInfra.relatedDeplymentPattern = currentUrl[currentUrl.length - 3];
```

Comment:
L63-L64 code logic is vulnerable for any change in URL contexts. It may fail silently. Is there a better way to handle this? Probably we can use some kind of regex or something similar to jax-rs @Path annotation value.

Reply:
This page will render when URL pattern matches to https://<host>:<port>/<product-ame>/<deployment-pattern-name>/test-plans/<testplan-id>.

Reply:
Ah ok..